src/components/Pressable/Pressable.tsx

```diff
@@ -27,412 +35,417 @@ const IS_TEST_ENV = isTestEnv();
 
 let IS_FABRIC: null | boolean = null;
 
-export default function Pressable(props: PressableProps) {
-  const {
-    testOnly_pressed,
-    hitSlop,
-    pressRetentionOffset,
-    delayHoverIn,
-    onHoverIn,
-    delayHoverOut,
-    onHoverOut,
-    delayLongPress,
-    unstable_pressDelay,
-    onPress,
-    onPressIn,
-    onPressOut,
-    onLongPress,
-    style,
-    children,
-    android_disableSound,
-    android_ripple,
-    disabled,
-    ...remainingProps
-  } = props;
-
-  const [pressedState, setPressedState] = useState(testOnly_pressed ?? false);
-
-  const pressableRef = useRef<View>(null);
-
-  // Disabled when onLongPress has been called
-  const isPressCallbackEnabled = useRef<boolean>(true);
-  const hasPassedBoundsChecks = useRef<boolean>(false);
-  const shouldPreventNativeEffects = useRef<boolean>(false);
-
-  const normalizedHitSlop: Insets = useMemo(
-    () =>
-      typeof hitSlop === 'number' ? numberAsInset(hitSlop) : (hitSlop ?? {}),
-    [hitSlop]
-  );
-
-  const normalizedPressRetentionOffset: Insets = useMemo(
-    () =>
-      typeof pressRetentionOffset === 'number'
-        ? numberAsInset(pressRetentionOffset)
-        : (pressRetentionOffset ?? {}),
-    [pressRetentionOffset]
-  );
-
-  const hoverInTimeout = useRef<number | null>(null);
-  const hoverOutTimeout = useRef<number | null>(null);
-
-  const hoverGesture = useMemo(
-    () =>
-      Gesture.Hover()
-        .manualActivation(true) // Stops Hover from blocking Native gesture activation on web
-        .cancelsTouchesInView(false)
-        .onBegin((event) => {
-          if (hoverOutTimeout.current) {
-            clearTimeout(hoverOutTimeout.current);
-          }
-          if (delayHoverIn) {
-            hoverInTimeout.current = setTimeout(
-              () => onHoverIn?.(gestureToPressableEvent(event)),
-              delayHoverIn
-            );
-            return;
-          }
-          onHoverIn?.(gestureToPressableEvent(event));
-        })
-        .onFinalize((event) => {
-          if (hoverInTimeout.current) {
-            clearTimeout(hoverInTimeout.current);
-          }
-          if (delayHoverOut) {
-            hoverOutTimeout.current = setTimeout(
-              () => onHoverOut?.(gestureToPressableEvent(event)),
-              delayHoverOut
-            );
-            return;
-          }
-          onHoverOut?.(gestureToPressableEvent(event));
-        }),
-    [delayHoverIn, delayHoverOut, onHoverIn, onHoverOut]
-  );
-
-  const pressDelayTimeoutRef = useRef<number | null>(null);
-  const isTouchPropagationAllowed = useRef<boolean>(false);
+const Pressable = forwardRef(
```

Comment:
We can also do something similar as in [buttons](https://github.com/software-mansion/react-native-gesture-handler/blob/324a236f7dc6d0234c878f35764b4f2938c37192/src/components/GestureButtons.tsx#L149)